streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -600,6 +609,75 @@ private void to(final TopicNameExtractor<K, V> topicExtractor,
         builder.addGraphNode(streamsGraphNode, sinkNode);
     }
 
+    @Override
+    public KTable<K, V> toTable() {
+        return toTable(NamedInternal.empty());
+    }
+
+    @Override
+    public KTable<K, V> toTable(final Named named) {
+        final ConsumedInternal<K, V> consumedInternal = new ConsumedInternal<>(Consumed.with(keySerde, valSerde));
+
+        final MaterializedInternal<K, V, KeyValueStore<Bytes, byte[]>> materializedInternal =
+            new MaterializedInternal<>(
+                Materialized.with(consumedInternal.keySerde(), consumedInternal.valueSerde()),
+                builder,
+                TO_KTABLE_NAME);
+
```

Comment:
We have to addStateStore to topology so we can join
so we needs to be materialized
internalStreamBuilder is similarly materialized.

KTableKTableJoinNode is process check  
-> 
if (!stateFactories.containsKey(stateStoreName)) {
            throw new TopologyException("StateStore " + stateStoreName + " is not added yet.");
   }

topologyBuilder.addStateStore(storeBuilder, nodeName());